README.md

```diff
@@ -1,2 +1,121 @@
+- **Unit Handling**: Translate OTLP units to Prometheus unit conventions
+- **Type-Aware Suffixes**: Optionally append `_total`, `_ratio` based on metric type
+- **Namespace Support**: Add configurable namespace prefixes
+- **UTF-8 Support**: Choose between Prometheus legacy scheme compliant metric/label names (`[a-zA-Z0-9:_]`) or untranslated metric/label names
+
+## Installation
+
+```bash
+go get github.com/prometheus/otlptranslator
+```
+
+## Quick Start
+
+```go
+package main
+
+import (
+    "fmt"
+    "github.com/prometheus/otlptranslator"
+)
+
+func main() {
+    // Create a metric namer
+    namer := otlptranslator.MetricNamer{
+        Namespace:          "myapp",
+        WithMetricSuffixes: true,
+        UTF8Allowed:        false,
+    }
+
+    // Convert OTLP metric to Prometheus format
+    metric := otlptranslator.Metric{
+        Name: "http.server.request.duration",
+        Unit: "s",
+        Type: otlptranslator.MetricTypeHistogram,
+    }
+    fmt.Println(namer.Build(metric)) // Output: myapp_http_server_request_duration_seconds
+
+    // Normalize label names
+    labelNamer := otlptranslator.LabelNamer{UTF8Allowed: false}
+    fmt.Println(labelNamer.Build("http.method")) // Output: http_method
+}
+```
+
+## Usage Examples
+
+### Metric Name Translation
+
+```go
+namer := otlptranslator.MetricNamer{WithMetricSuffixes: true, UTF8Allowed: false}
+
+// Counter gets _total suffix
+counter := otlptranslator.Metric{
+    Name: "requests.count", Unit: "1", Type: otlptranslator.MetricTypeMonotonicCounter,
+}
+fmt.Println(namer.Build(counter)) // requests_count_total
+
+// Gauge with unit conversion
+gauge := otlptranslator.Metric{
+    Name: "memory.usage", Unit: "By", Type: otlptranslator.MetricTypeGauge,
+}
+fmt.Println(namer.Build(gauge)) // memory_usage_bytes
+
+// Dimensionless gauge gets _ratio suffix
+ratio := otlptranslator.Metric{
+    Name: "cpu.utilization", Unit: "1", Type: otlptranslator.MetricTypeGauge,
+}
+fmt.Println(namer.Build(ratio)) // cpu_utilization_ratio
+```
+
+### Label Translation
+
+```go
+labelNamer := otlptranslator.LabelNamer{UTF8Allowed: false}
+
+labelNamer.Build("http.method")           // http_method
+labelNamer.Build("123invalid")            // key_123invalid  
+labelNamer.Build("_private")              // key_private
+labelNamer.Build("__reserved__")          // __reserved__ (preserved)
+labelNamer.Build("label@with$symbols")    // label_with_symbols
+```
+
+### Unit Translation
+
+```go
+unitNamer := otlptranslator.UnitNamer{UTF8Allowed: false}
+
+unitNamer.Build("s")           // seconds
+unitNamer.Build("By")          // bytes
+unitNamer.Build("requests/s")  // requests_per_second
+unitNamer.Build("1")           // "" (dimensionless)
+```
+
+### Configuration Options
+
+```go
+// Prometheus-compliant mode (default) - supports [a-zA-Z0-9:_]
+compliantNamer := otlptranslator.MetricNamer{UTF8Allowed: false}
```

Comment:
```suggestion
// Prometheus-compliant mode (default) - supports [a-zA-Z0-9:_]
compliantNamer := otlptranslator.MetricNamer{WithMetricSuffixes: true, UTF8Allowed: false}
```

Reply:
technically the default config is Escaping with no suffixes... do we want to mention that?